docs/ossm/ossm2-migration/cluster-wide/README.md

```diff
@@ -0,0 +1,133 @@
+# OpenShift Service Mesh 2.6 Cluster wide --> 3 Migration guide
+This guide is for users who are currently running `ClusterWide` OpenShift Service Mesh 2.6 migrating to OpenShift Service Mesh 3.0. You should first read [this document comparing OpenShift Service Mesh 2 vs. OpenShift Service Mesh 3](../../ossm2-vs-ossm3.md) to familiarize yourself with the concepts between the two versions and the differences in managing the workloads and addons.
+
+## Migrating OpenShift Service Mesh 2.6 Cluster wide to OpenShift Service Mesh 3
+
+### Prerequisites
+- you have completed all the steps from the [pre-migration checklist](../README.md#pre-migration-checklist)
+- you have read [OpenShift Service Mesh 2 vs. OpenShift Service Mesh 3](../../ossm2-vs-ossm3.md)
+- you have verified that your 2.6 `ServiceMeshControlPlane` is using `ClusterWide` mode
+- Red Hat OpenShift Service Mesh 3 operator is installed
+- `IstioCNI` is installed
+- `istioctl` is installed
+
+### Procedure
+In this example, we'll be using the [bookinfo demo](https://raw.githubusercontent.com/Maistra/istio/maistra-2.6/samples/bookinfo/platform/kube/bookinfo.yaml) but you can follow these same steps with your own workloads.
+
+#### Plan the migration
+There will be two cluster wide Istio control planes running during the migration process so it's necessary to plan the migration steps in advance in order to avoid possible conflicts between the two control planes.
+
+There is a few conditions which must be assured to achieve the seamless migration:
+- both control planes must share the same root certificate
+
+  This can be achieved simply by installing the 3.0 control plane to the same namespace as 2.6 control plane
+- 3.0 control plane must have access to all namespaces in 2.6 mesh
+
+  At some point during the migration, some proxies will be already controlled by the 3.0 control plane and some still controlled by the 2.6 control plane. To assure the communication still works, both control planes must be aware of the same set of services. You have to verify that:
+  1. there are no Network Policies blocking the traffic
+  1. if used, `discoverySelectors`  defined in 3.0 `Istio` include all namespaces from 2.6 mesh. See [Scoping the service mesh with DiscoverySelectors](../../create-mesh/README.md)
+- only one control plane will try to inject a side car
+
+  This can be achieved by correct use of injection labels. Please see [Installing the Sidecar](../../injection/README.md) for details.
+  > **_NOTE:_** To disable 2.6 injector, we will use `maistra.io/ignore-namespace: "true"` label in this guide.
+
+Apart from the conditions above, it's recommended to decide which injection labels will be used. See [Installing the Sidecar](../../injection/README.md) explaining relation between Istio revisions and injection labels. Based on the decision, use correct injection labels later in this guide.
+
+#### Create your Istio resource
+See Installation documentation for details.
```

Comment:
I think it's good to show a more full example here, including creation of the `Istio` resource, as you mentioned some requirements above like it needs to be in the name namespace as the OSSM 2 control plane, and the discovery selectors should match the namespaces of the OSSM 2 meshes. So, as this is a pretty big procedure, I think this could be a more complete end to end example.

Reply:
I added back the Istio CR creation part. Just to avoid confusion, is this the final decision to keep it there? Asking because it was removed due to previous review comment: "rather than giving specific instructions to create a new Istio resource here, we could direct them to the OSSM installation instructions, which covers discoverySelectors."

Reply:
Yes, sorry for the back and forth... I guess after seeing it, it's better to include it.